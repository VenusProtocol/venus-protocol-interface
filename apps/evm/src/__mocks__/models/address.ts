const address = '0x3d759121234cd36F8124C21aFe1c6852d2bEd848';

export const altAddress = '0xa258a693A403b7e98fd05EE9e1558C760308cFC7';

export default address;
