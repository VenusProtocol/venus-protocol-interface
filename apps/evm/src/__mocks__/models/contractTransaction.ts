import { ChainId } from 'types';
import type { Transaction } from 'viem';

const contractTransaction: Transaction = {
  hash: '0xfake-transaction-hash',
  type: 'legacy',
  from: '0x1d759121234cd36F8124C21aFe1c6852d2bEd848',
  to: '0x2d759121234cd36F8124C21aFe1c6852d2bEd834',
  nonce: 10,
  value: 0n,
  chainId: ChainId.BSC_TESTNET,
  blockHash: '0xfake-block-hash',
  blockNumber: 1123213n,
  gas: 12n,
  gasPrice: 1n,
  input: '0xfake-input',
  r: '0xfake-r',
  s: '0xfake-s',
  v: 3n,
  transactionIndex: 1,
  typeHex: '0xfake-type-hex',
};

export default contractTransaction;
