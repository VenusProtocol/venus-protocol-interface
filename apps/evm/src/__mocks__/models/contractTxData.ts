import type { ContractTxData } from 'types';

const contractTxData: ContractTxData<any, any> = {
  contract: {},
  methodName: 'fakeMethod',
  args: [],
};

export default contractTxData;
