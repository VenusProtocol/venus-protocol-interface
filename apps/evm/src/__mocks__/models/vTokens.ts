import type { VToken } from 'types';

import { bnb, busd, eth, lisUsd, luna, usdc, usdt, ust, wbnb, weth, xvs } from './tokens';

export const vXvs: VToken = {
  address: '0x6d6F697e34145Bb95c54E77482d97cc261Dc237E',
  decimals: 8,
  symbol: 'vXVS',
  underlyingToken: xvs,
};

export const vBnb: VToken = {
  address: '0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c',
  decimals: 8,
  symbol: 'vBNB',
  underlyingToken: bnb,
};

export const vUsdc: VToken = {
  address: '0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7',
  decimals: 8,
  symbol: 'vUSDC',
  underlyingToken: usdc,
};

export const vUsdtCorePool: VToken = {
  address: '0xb7526572FFE56AB9D7489838Bf2E18e3323b441A',
  decimals: 8,
  symbol: 'vUSDT',
  underlyingToken: usdt,
};

export const vUsdtDeFi: VToken = {
  address: '0x80CC30811e362aC9aB857C3d7875CbcCc0b65750',
  decimals: 8,
  symbol: 'vUSDT_DeFi',
  underlyingToken: usdt,
};

export const vUsdtGameFi: VToken = {
  address: '0x0bFE4e0B8A2a096A27e5B18b078d25be57C08634',
  decimals: 8,
  symbol: 'vUSDT_GameFi',
  underlyingToken: usdt,
};

export const vUsdtLiquidStakedBnb: VToken = {
  address: '0x2197d02cC9cd1ad51317A0a85A656a0c82383A7c',
  decimals: 8,
  symbol: 'vUSDT_Liquid_Staked_BNB',
  underlyingToken: usdt,
};

export const vUsdtMeme: VToken = {
  address: '0x3AF2bE7AbEF0f840b196D99d79F4B803a5dB14a1',
  decimals: 8,
  symbol: 'vUSDT_Meme',
  underlyingToken: usdt,
};

export const vUsdtStablecoins: VToken = {
  address: '0x3338988d0beb4419Acb8fE624218754053362D06',
  decimals: 8,
  symbol: 'vUSDT_Stablecoins',
  underlyingToken: usdt,
};

export const vUsdtTron: VToken = {
  address: '0x712774CBFFCBD60e9825871CcEFF2F917442b2c3',
  decimals: 8,
  symbol: 'vUSDT_Tron',
  underlyingToken: usdt,
};

export const vBusdCorePool: VToken = {
  address: '0x08e0A5575De71037aE36AbfAfb516595fE68e5e4',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vBusdStablecoins: VToken = {
  address: '0x8c8A1a0b6e1cb8058037F7bF24de6b79Aca5B7B0',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vUst: VToken = {
  address: '0xF206af85BC2761c4F876d27Bd474681CfB335EfA',
  decimals: 8,
  symbol: 'vUST',
  underlyingToken: ust,
};

export const vLuna: VToken = {
  address: '0x9C3015191d39cF1930F92EB7e7BCbd020bCA286a',
  decimals: 8,
  symbol: 'vLUNA',
  underlyingToken: luna,
};

export const vWeth: VToken = {
  address: '0xc2931B1fEa69b6D6dA65a50363A8D75d285e4da9',
  decimals: 8,
  symbol: 'vWETH',
  underlyingToken: weth,
};

export const vEth: VToken = {
  address: '0x86f8DfB7CA84455174EE9C3edd94867b51Da46BD',
  decimals: 8,
  symbol: 'vETH',
  underlyingToken: eth,
};

export const vLisUSD: VToken = {
  address: '0x170d3b2da05cc2124334240fB34ad1359e34C562',
  decimals: 8,
  symbol: 'vlisUSD',
  underlyingToken: lisUsd,
};

export const vWbnb: VToken = {
  address: '0x231dED0Dfc99634e52EE1a1329586bc970d773b3',
  decimals: 8,
  symbol: 'vWBNB',
  underlyingToken: wbnb,
};

export default [
  vXvs,
  vBnb,
  vUsdc,
  vUsdtCorePool,
  vUsdtDeFi,
  vUsdtGameFi,
  vUsdtLiquidStakedBnb,
  vUsdtMeme,
  vUsdtStablecoins,
  vUsdtTron,
  vBusdCorePool,
  vBusdStablecoins,
  vUst,
  vLuna,
  vEth,
  vLisUSD,
  vWbnb,
];
