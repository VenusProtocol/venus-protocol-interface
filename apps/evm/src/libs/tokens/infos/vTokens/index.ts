import type { VTokenAssets } from 'libs/tokens/types';
import { ChainId } from 'types';
import { vTokenAssets as arbitrumOneVTokenAssets } from './arbitrumOne';
import { vTokenAssets as arbitrumSepoliaVTokenAssets } from './arbitrumSepolia';
import { vTokenAssets as baseMainnetVTokenAssets } from './baseMainnet';
import { vTokenAssets as baseSepoliaVTokenAssets } from './baseSepolia';
import { vTokenAssets as berachainBepoliaVTokenAssets } from './berachainBepolia';
import { vTokenAssets as berachainMainnetVTokenAssets } from './berachainMainnet';
import { vTokenAssets as bscMainnetVTokenAssets } from './bscMainnet';
import { vTokenAssets as bscTestnetVTokenAssets } from './bscTestnet';
import { vTokenAssets as ethereumVTokenAssets } from './ethereum';
import { vTokenAssets as opBnbMainnetVTokenAssets } from './opBnbMainnet';
import { vTokenAssets as opBnbTestnetVTokenAssets } from './opBnbTestnet';
import { vTokenAssets as optimismMainnetVTokenAssets } from './optimismMainnet';
import { vTokenAssets as optimismSepoliaVTokenAssets } from './optimismSepolia';
import { vTokenAssets as sepoliaVTokenAssets } from './sepolia';
import { vTokenAssets as unichainMainnetVTokenAssets } from './unichainMainnet';
import { vTokenAssets as unichainSepoliaVTokenAssets } from './unichainSepolia';
import { vTokenAssets as zkSyncMainnetVTokenAssets } from './zkSyncMainnet';
import { vTokenAssets as zkSyncSepoliaVTokenAssets } from './zkSyncSepolia';

export const vTokenAssetsPerChainId: Record<ChainId, VTokenAssets> = {
  [ChainId.BSC_MAINNET]: bscMainnetVTokenAssets,
  [ChainId.BSC_TESTNET]: bscTestnetVTokenAssets,
  [ChainId.OPBNB_MAINNET]: opBnbMainnetVTokenAssets,
  [ChainId.OPBNB_TESTNET]: opBnbTestnetVTokenAssets,
  [ChainId.ETHEREUM]: ethereumVTokenAssets,
  [ChainId.SEPOLIA]: sepoliaVTokenAssets,
  [ChainId.ARBITRUM_ONE]: arbitrumOneVTokenAssets,
  [ChainId.ARBITRUM_SEPOLIA]: arbitrumSepoliaVTokenAssets,
  [ChainId.ZKSYNC_MAINNET]: zkSyncMainnetVTokenAssets,
  [ChainId.ZKSYNC_SEPOLIA]: zkSyncSepoliaVTokenAssets,
  [ChainId.OPTIMISM_MAINNET]: optimismMainnetVTokenAssets,
  [ChainId.OPTIMISM_SEPOLIA]: optimismSepoliaVTokenAssets,
  [ChainId.BASE_MAINNET]: baseMainnetVTokenAssets,
  [ChainId.BASE_SEPOLIA]: baseSepoliaVTokenAssets,
  [ChainId.UNICHAIN_MAINNET]: unichainMainnetVTokenAssets,
  [ChainId.UNICHAIN_SEPOLIA]: unichainSepoliaVTokenAssets,
  [ChainId.BERACHAIN_MAINNET]: berachainMainnetVTokenAssets,
  [ChainId.BERACHAIN_BEPOLIA]: berachainBepoliaVTokenAssets,
};
