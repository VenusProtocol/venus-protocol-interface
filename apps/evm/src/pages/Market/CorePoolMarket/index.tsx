/** @jsxImportSource @emotion/react */
import { useParams } from 'react-router-dom';

import { useGetChainMetadata } from 'hooks/useGetChainMetadata';

import MarketLoader from 'containers/MarketLoader';
import { Market } from '../Market';

const CorePoolMarket: React.FC = () => {
  const { vTokenAddress } = useParams();
  const { corePoolComptrollerContractAddress } = useGetChainMetadata();

  return (
    <MarketLoader
      poolComptrollerAddress={corePoolComptrollerContractAddress}
      vTokenAddress={vTokenAddress}
    >
      {marketProps => <Market {...marketProps} />}
    </MarketLoader>
  );
};

export default CorePoolMarket;
