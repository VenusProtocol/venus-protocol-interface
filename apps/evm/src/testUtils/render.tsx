import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render as renderComponentTl } from '@testing-library/react';
import { renderHook as renderHookTl } from '@testing-library/react-hooks';
import type { ReactElement } from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import type Vi from 'vitest';

import fakeSigner from '__mocks__/models/signer';

import { Web3Wrapper, useAccountAddress, useChainId, useSigner } from 'libs/wallet';
import { MuiThemeProvider } from 'theme/MuiThemeProvider';
import type { ChainId } from 'types';

const createQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        gcTime: 0,
      },
    },
  });

interface Options {
  accountAddress?: string;
  chainId?: ChainId;
  routerInitialEntries?: string[];
  routePath?: string;
}

interface WrapperProps {
  queryClient: QueryClient;
  children?: React.ReactNode;
  options?: Partial<Options>;
}

const Wrapper: React.FC<WrapperProps> = ({ children, queryClient, options }) => {
  if (options?.accountAddress) {
    const accountAddress = options?.accountAddress;

    (useAccountAddress as Vi.Mock).mockImplementation(() => ({
      accountAddress,
    }));

    (useSigner as Vi.Mock).mockImplementation(() => ({
      signer: {
        ...fakeSigner,
        getAddress: async () => accountAddress,
      },
    }));
  }

  if (options?.chainId) {
    (useChainId as Vi.Mock).mockImplementation(() => ({
      chainId: options?.chainId,
    }));
  }

  return (
    <MuiThemeProvider>
      <QueryClientProvider client={queryClient}>
        <Web3Wrapper>
          <MemoryRouter initialEntries={options?.routerInitialEntries || ['/']}>
            <Routes>
              <Route path={options?.routePath || '/'} element={children} />
            </Routes>
          </MemoryRouter>
        </Web3Wrapper>
      </QueryClientProvider>
    </MuiThemeProvider>
  );
};

export const renderComponent = (children: ReactElement, options?: Partial<Options>) => {
  const queryClient = createQueryClient();

  const renderRes = renderComponentTl(children, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};

export const renderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  options?: Partial<Options>,
) => {
  const queryClient = createQueryClient();

  const renderRes = renderHookTl(hook, {
    wrapper: props => <Wrapper queryClient={queryClient} options={options} {...props} />,
  });

  return {
    ...renderRes,
    queryClient,
  };
};
