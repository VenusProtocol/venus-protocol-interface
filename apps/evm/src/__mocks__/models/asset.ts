import BigNumber from 'bignumber.js';

import { busd, usdc, usdt, xvs } from '__mocks__/models/tokens';
import { vBusd, vUsdc, vUsdt, vXvs } from '__mocks__/models/vTokens';

import { Asset } from 'types';

export const assetData: Asset[] = [
  {
    vToken: vXvs,
    supplyApyPercentage: new BigNumber('0.05225450324405023'),
    borrowApyPercentage: new BigNumber('-2.3062487835658776'),
    collateralFactor: 0.5,
    reserveFactor: 0.25,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('127.86734'),
    liquidityCents: new BigNumber(8036465875),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(49.589181233),
    userWalletBalanceTokens: new BigNumber('100'),
    userWalletBalanceCents: new BigNumber('12786'),
    userSupplyBalanceTokens: new BigNumber('90'),
    userSupplyBalanceCents: new BigNumber('11508'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: true,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(70925716),
    supplyBalanceCents: new BigNumber(278311516),
    supplyBalanceTokens: new BigNumber('19339683254955736'),
    borrowBalanceTokens: new BigNumber('1852935.597521220541385584'),
    supplyPercentageRatePerBlock: new BigNumber('0.000001'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000001'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.11720675342484096'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('4.17469243006608279'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
  },
  {
    vToken: vUsdc,
    supplyApyPercentage: new BigNumber('3.887242555711379188'),
    borrowApyPercentage: new BigNumber('-5.361233028654066'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('99.99364'),
    liquidityCents: new BigNumber(1702951959),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1),
    userWalletBalanceTokens: new BigNumber('0'),
    userWalletBalanceCents: new BigNumber('0'),
    userSupplyBalanceTokens: new BigNumber('100'),
    userSupplyBalanceCents: new BigNumber('9999'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(858721657509436),
    supplyBalanceCents: new BigNumber(1000183891879506),
    supplyBalanceTokens: new BigNumber('47171999131879271527200010830'),
    borrowBalanceTokens: new BigNumber('73128318.509651061457900627'),
    supplyPercentageRatePerBlock: new BigNumber('0.000002'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000002'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.353105649796123742'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('0.953105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.753105649796123742'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.670327607690572731'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('1.233105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
  {
    vToken: vUsdt,
    supplyApyPercentage: new BigNumber('3.593608909332766999'),
    borrowApyPercentage: new BigNumber('-4.9748661428011145'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.024602'),
    liquidityCents: new BigNumber(5534102886),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(0.981982),
    userWalletBalanceTokens: new BigNumber('900'),
    userWalletBalanceCents: new BigNumber('90000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('40'),
    userBorrowBalanceCents: new BigNumber('4000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(3158444721),
    supplyBalanceCents: new BigNumber(1098041201011568),
    supplyBalanceTokens: new BigNumber('5029972090817266864401527367893625'),
    borrowBalanceTokens: new BigNumber('232511166.920938849475104194'),
    supplyPercentageRatePerBlock: new BigNumber('0.000003'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000003'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.421719501189155143'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.522209972682294832'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.015310564979612374'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
  },
  {
    vToken: vBusd,
    supplyApyPercentage: new BigNumber('2.886396363044176106'),
    borrowApyPercentage: new BigNumber('-4.050271277344538'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.000922'),
    liquidityCents: new BigNumber(3654492935),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1.000003),
    userWalletBalanceTokens: new BigNumber('110'),
    userWalletBalanceCents: new BigNumber('11000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('50'),
    userBorrowBalanceCents: new BigNumber('5000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(83910350502),
    supplyBalanceCents: new BigNumber(1054707853878),
    supplyBalanceTokens: new BigNumber('51881081291203672464'),
    borrowBalanceTokens: new BigNumber('142662020.229587308931217432'),
    supplyPercentageRatePerBlock: new BigNumber('0.000004'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000004'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.678420831753642169'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: xvs,
        apyPercentage: new BigNumber(0),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.852697602175970714'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: busd,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
];
