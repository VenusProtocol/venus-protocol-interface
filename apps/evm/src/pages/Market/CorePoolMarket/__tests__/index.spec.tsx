import { waitFor } from '@testing-library/react';
import type Vi from 'vitest';

import { assetData } from '__mocks__/models/asset';
import { vTokenApySimulations } from '__mocks__/models/vTokenApySimulations';
import { vXvs } from '__mocks__/models/vTokens';
import { renderComponent } from 'testUtils/render';

import {
  getVTokenApySimulations,
  useGetAsset,
  useGetIsolatedPoolVTokenLiquidationThreshold,
  useGetPoolLiquidationIncentive,
} from 'clients/api';

import CorePoolMarket from '..';
import TEST_IDS from '../../testIds';

describe('CorePoolMarket', () => {
  beforeEach(() => {
    (useGetAsset as Vi.Mock).mockImplementation(() => ({
      isLoading: false,
      data: {
        asset: assetData[0],
      },
    }));

    (getVTokenApySimulations as Vi.Mock).mockImplementation(() => ({
      apySimulations: vTokenApySimulations,
    }));

    (useGetPoolLiquidationIncentive as Vi.Mock).mockImplementation(() => ({
      isLoading: false,
      data: {
        liquidationIncentivePercentage: 10,
      },
    }));

    (useGetIsolatedPoolVTokenLiquidationThreshold as Vi.Mock).mockImplementation(() => ({
      isLoading: false,
      data: {
        liquidationThresholdPercentage: 80,
      },
    }));
  });

  it('renders without crashing', () => {
    renderComponent(<CorePoolMarket />, {
      routerInitialEntries: [`/${vXvs.address}`],
      routePath: '/:vTokenAddress',
    });
  });

  it('fetches market details and displays them correctly', async () => {
    const { getByTestId, queryByTestId } = renderComponent(<CorePoolMarket />, {
      routerInitialEntries: [`/${vXvs.address}`],
      routePath: '/:vTokenAddress',
    });

    // Check interest rate model displays correctly
    await waitFor(() =>
      expect(getByTestId(TEST_IDS.interestRateModel).textContent).toMatchSnapshot(),
    );
    // Check supply info is not displayed
    expect(queryByTestId(TEST_IDS.supplyInfo)).toBeNull();
    // Check borrow info is not displayed
    expect(queryByTestId(TEST_IDS.borrowInfo)).toBeNull();
    // Check market info displays correctly
    expect(getByTestId(TEST_IDS.marketInfo).textContent).toMatchSnapshot();
  });
});
