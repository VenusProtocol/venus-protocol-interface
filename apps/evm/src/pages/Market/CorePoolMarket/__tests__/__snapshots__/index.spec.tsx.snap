// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorePoolMarket > fetches market details and displays them correctly 1`] = `"Interest Rate ModelUtilization rateBorrow APYSupply APY"`;

exports[`CorePoolMarket > fetches market details and displays them correctly 2`] = `"Supply info0%Total supplied> $100T / > $100T> 100T / > 100T XVSAverage APY-Current APY0.16%"`;

exports[`CorePoolMarket > fetches market details and displays them correctly 3`] = `"Borrow info< 0.01%Total borrowed$2.36M / > $100T1.85M / > 100T XVSAverage APY-Current APY-6.48%Liquidation Threshold50%Liquidation Penalty10%"`;

exports[`CorePoolMarket > fetches market details and displays them correctly 4`] = `"Market infoDaily supplying interests$3.98Daily borrowing interests-$44.81Daily XVS distributed19.99MReserve factor25%Exchange rate1 XVS=49.589181 vXVS"`;
